frontend/libs/studio-components-legacy/src/components/StudioExpression/StudioExpressionContext.tsx

```diff
@@ -0,0 +1,35 @@
+import type { ComponentProps, Context } from 'react';
+import React, { createContext, useContext } from 'react';
+import type { DataLookupOptions } from './types/DataLookupOptions';
+import type { ExpressionTexts } from './types/ExpressionTexts';
+import type { SimpleSubexpressionValueType } from './enums/SimpleSubexpressionValueType';
+
+export type StudioExpressionContext = {
+  dataLookupOptions: DataLookupOptions;
+  texts: ExpressionTexts;
+  types: SimpleSubexpressionValueType[];
+};
+
+type NullableStudioExpressionContext = StudioExpressionContext | null;
+
+const StudioExpressionContext = createContext<NullableStudioExpressionContext>(null);
+
+export type StudioExpressionContextProviderProps = ComponentProps<
+  Context<StudioExpressionContext>['Provider']
+>;
+
+export function StudioExpressionContextProvider(
+  props: StudioExpressionContextProviderProps,
+): React.ReactElement {
+  return <StudioExpressionContext.Provider {...props} />;
+}
+
+export const useStudioExpressionContext = (): StudioExpressionContext => {
+  const context = useContext<NullableStudioExpressionContext>(StudioExpressionContext);
+  if (context === null) {
+    throw new Error(
+      'useStudioExpressionContext must be used within StudioExpressionContextProvider.',
+    );
+  }
```

Comment:
Code coverage fails because this error case is never reached, which is intended. We could make the check pass by creating a test that isolates this hook, but such a test would not provide any real value other than making Codecov happy. We know that the hook works because many of the `StudioExpression` tests would not pass without it, and it's only meant to be used internally in this component.